src/components/CustomFileManager.vue

```diff
@@ -310,113 +315,175 @@ export default class CustomFileManager extends Vue {
 
   async itemToChips(selectable: IGirderSelectAble) {
     const ret: IChipAttrs[] = [];
-    const baseChip = {
-      color: "blue",
+
+    // Determine type first
+    const isDataset = isDatasetFolder(selectable);
+    const isConfig = isConfigurationItem(selectable);
+
+    if (!isDataset && !isConfig) {
+      return { chips: ret, type: null as any };
+    }
+
+    const type = isDataset ? "dataset" : "configuration";
+
+    // First chip (type indicator)
+    const headerChip: IChipAttrs = {
+      text: isDataset ? "Dataset" : "Collection",
+      color: "grey darken-1",
     };
+    if (this.clickableChips) {
+      headerChip.to = isDataset
+        ? { name: "dataset", params: { datasetId: selectable._id } }
+        : {
+            name: "configuration",
+            params: { configurationId: selectable._id },
+          };
+    }
+    ret.push(headerChip);
 
-    // Store the type with the chips
-    let itemType = null;
+    // If we don't show more chips, return early
+    if (!this.moreChips) {
+      return { chips: ret, type };
+    }
 
-    // Add chips if item is a dataset
-    if (isDatasetFolder(selectable)) {
-      itemType = "dataset";
-      // Dataset chip
-      const firstChip: IChipAttrs = {
-        text: "Dataset",
-        color: "grey darken-1",
+    // Enqueue for batch resolution and return header-only chips for now
+    if (isDataset) {
+      this.batchQueueDatasetIds.add(selectable._id);
+    } else {
+      this.batchQueueConfigurationIds.add(selectable._id);
+    }
+    this.scheduleBatchResolve();
+
+    return { chips: ret, type };
+  }
+
+  scheduleBatchResolve() {
+    if (this.batchTimer !== null) {
+      window.clearTimeout(this.batchTimer);
+      this.batchTimer = null;
+    }
+    // Debounce to accumulate ids from the current render pass
+    this.batchTimer = window.setTimeout(() => this.flushBatchResolve(), 50);
+  }
```

Comment:
I get why (well, it took me a while ^^') GPT would chose this implementation, but damn it's complex.  Could you add a few comments explaining that this timer allows the multiple calls to `itemToChips` to gather dataset and configuration ids before firing a single call to `flushBatchResolve` ?